### DISCUSSION TITLE
Ctrl-v key bindings don't work in Windows Terminal

### DISCUSSION THREAD

**Discussion by rowanlovejoy** [helix-editor/helix#10295] [General] [answered]
### Summary

The key binding for opening a file in a new vertical split (Ctrl-v) from the file picker doesn't work. The corresponding key binding for opening a file in a new horizontal split (Ctrl-s) does work. In Window mode, the key binding “v” for opening a new vertical split does work, but the Ctrl-v key binding doesn't. 

I've noticed that if my clipboard contains text, if I press Ctrl-v in the file picker, it pastes the text from my clipboard into the fuzzy finder. This also happens if I use the Window mode Ctrl-v binding, my clipboard contents being pasted into the file. I think what might be happening is that the Windows paste key binding, which is also Ctrl-v, is overriding Helix's.

This issue occurs using any shell inside Windows Terminal. I have tried PowerShell, Command Prompt, and Git Bash. 

My helix.log file is empty.

Using WezTerm on the same machine, the Ctrl-v key bindings work correctly, even if my clipboard contains text.

### Reproduction Steps

 

I tried this:

1. Open the Windows Terminal
2. Navigate to a directory containing one or more files
3. Run 'hx' command
4. Open the file picker and select a file
5. Press Ctrl-v

I expected this to happen:

The selected file is opened in a new vertical split

Instead, this happened:

If my clipboard is empty, nothing happens. If my clipboard contains text, it is pasted into the fuzzy finder field.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

Windows 10 (22H2)

### Terminal Emulator

Windows Terminal 1.19.10821.0

### Installation Method

Winget

### Helix Version

helix 24.3 (2cadec0b)

**Comment by pppkin** [comment, marked as answer]
> the Windows paste key binding, which is also Ctrl-v, is overriding Helix's.

this. As far as helix concerns, ctrl-v never happened. It was processed before helix(or any terminal program) knows what's going on. This is not a bug on our side :)

**Reply by rowanlovejoy** [reply]
Thanks. I see that I answered my own question 😅. I've changed the Windows Terminal paste key binding, and now Helix's Ctrl-v bindings work correctly.

Windows Terminal is on the [list of terminals confirmed working with Helix](https://github.com/helix-editor/helix/wiki/Terminal-Support#windows-terminal) with no qualifications (except for its lack of Enhanced Keyboard Protocol). I think it would be worth adding a line or two under the Windows Terminal section stating that its default key bindings conflict with Helix.

Do you think this would be reasonable?

**Reply by pppkin** [reply]
Or we could mention this at the top the keymap document. Any terminal emulators, terminal multiplexers(for example, zellij) can conflict with our default keymap. There's no much we can do about it really.

**Reply by rowanlovejoy** [reply]
That would also be good, A generic “default key bindings may conflict” message at the top of the keymap document paired with call-outs for known conflicts on the wiki page would probably provide enough coverage to prevent other people getting confused like I was.

**Reply by rowanlovejoy** [reply]
I've gone ahead and made a PR for the docs change you suggested: https://github.com/helix-editor/helix/pull/10380#issue-2241401600 And I've also made an edit to the wiki to call Ctrl-v mapping conflict in Windows Terminal: https://github.com/helix-editor/helix/wiki/Terminal-Support#key-mappings
